firefox-ios/Client/Frontend/Settings/Main/AppSettingsTableViewController.swift

```diff
@@ -456,10 +456,12 @@ class AppSettingsTableViewController: SettingsTableViewController,
     // MARK: - UITableViewDelegate
 
     override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
-        let headerView = super.tableView(
+        guard let headerView = super.tableView(
             tableView,
             viewForHeaderInSection: section
-        ) as! ThemedTableSectionHeaderFooterView
+        ) as? ThemedTableSectionHeaderFooterView else {
+            fatalError("Failed to dequeue ThemedTableSectionHeaderFooterView for section \(section)")
```

Comment:
I'm not sure that this is the best approach, returning `fatalErrors` for tableViews.... @thatswinnie what do you think?

Reply:
Hi @adudenamedruby and @thatswinnie, I’ve made a change based on your feedback regarding the use of `fatalError` in the table view handling. Please let me know if this approach works better.